src/store/footer.rs

```diff
@@ -26,15 +27,18 @@ impl BinarySerializable for DocStoreFooter {
 
     fn deserialize<R: io::Read>(reader: &mut R) -> io::Result<Self> {
         let doc_store_version = u32::deserialize(reader)?;
-        if doc_store_version != DOC_STORE_VERSION {
-            panic!("actual doc store version: {doc_store_version}, expected: {DOC_STORE_VERSION}");
+        if doc_store_version > DOC_STORE_VERSION {
```

Comment:
can we introduce a `DocStoreVersion` enum?

Reply:
yep, I converted it to an enum